src/snowcli/cli/nativeapp/teardown_processor.py

```diff
@@ -0,0 +1,233 @@
+            if show_versions_cursor.rowcount > 0:
+                raise CouldNotDropApplicationPackageWithVersions()
+
+            # 4. Check distribution of the existing app pkg
+            actual_distribution = self.get_app_pkg_distribution_in_snowflake
+            if self.is_app_pkg_distribution_mismatched():
+                log.warning(
+                    dedent(
+                        f"""\
+                        Continuing to execute `snow app teardown` on app pkg {self.package_name} with distribution {actual_distribution}.
+                    """
+                    )
+                )
+
+            # 5. If distribution is internal, check for special comment
+            row_comment = show_obj_row[COMMENT_COL]
+            if actual_distribution == INTERNAL_DISTRIBUTION:
+                if row_comment == SPECIAL_COMMENT:
+                    self.needs_confirm = False
+                else:
+                    if self.needs_confirm and not auto_yes:
+                        log.warn(
+                            f"Application package {self.package_name} was not created by SnowCLI!"
+                        )
+            else:
+                if self.needs_confirm and not auto_yes:
+                    log.warn(
+                        f"Application package {self.package_name} in your Snowflake account has distribution property '{EXTERNAL_DISTRIBUTION}'"
+                    )
+
+            if self.needs_confirm and not auto_yes:
+                log.warn(
+                    dedent(
+                        f"""\
+                        Application package details:
+                        Name: {self.app_name}
+                        Created on: {show_obj_row["created_on"]}
+                        Distribution: {actual_distribution}
+                        Owner: {show_obj_row[OWNER_COL]}
+                        Comment: {show_obj_row[COMMENT_COL]}
+                    """
+                    )
+                )
+                should_drop_object = typer.confirm("Are you sure you want to it?")
+                if not should_drop_object:
+                    log.info(
+                        dedent(
+                            f"""\
+                                Did not drop application package {self.package_name}.
+                            """
+                        )
+                    )
+                    return  # The user desires to keep the app pkg, therefore exit gracefully
+
+            log.info(f"Dropping application package {self.package_name} now.")
+            drop_query = f"drop application package {self.package_name}"
+            try:
+                self._execute_query(drop_query)
+            except:
+                raise SnowflakeSQLExecutionError(drop_query)
+
+            log.info(f"Dropped application package {self.package_name} successfully.")
+            return True
+
+    def process(self, auto_yes: bool = False, *args, **kwargs):
+
+        self.needs_confirm = True
```

Comment:
Because both drop procedures have independent `needs_confirm`, I think it's safest to push this down from being an instance property and just have it as a local variable in each method

Reply:
Addressed in the next commit.